sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Content/ContainerRegistryContentClient.cs

```diff
@@ -532,10 +536,27 @@ private static string GetContentRange(long offset, long length)
             return FormattableString.Invariant($"{offset}-{endRange}");
         }
 
-        private static long GetBlobLengthFromContentRange(string contentRange)
+        private static long GetBlobSize(Response response)
         {
-            string size = contentRange.Split('/')[1];
-            return long.Parse(size, CultureInfo.InvariantCulture);
+            if (!response.Headers.TryGetValue("Content-Range", out string contentRange) ||
+                contentRange == null)
+            {
+                throw new RequestFailedException(response.Status, InvalidContentRangeMessage);
+            }
+
+            string[] values = contentRange.Split('/');
```

Comment:
nit: it's a bit cheaper to do `IndexOf('\')` and substring the seconds part

Reply:
TY!